### PR TITLE
Do not render ChannelHeaderMobile when not in mobile view to prevent heavy selector computations

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
The NotifyCounts component which is mounted as a child component of the tree under ChannelHeaderMobile uses a very heavy selector `getUnreadStatusInCurrentTeam` that recomputes often. Since the mobile channel header is only needed when we're in mobile view, this PR changes it to only mount when we're looking at mobile view.

@jgilliam17 for testing this one can you make sure that the mobile channel header still appears when loading the app in a mobile browser or when shrinking the app in a desktop browser to be small?

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by M-ZubairAhmed — APPROVED** [review]

File: components/channel_layout/center_channel/index.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ const mapStateToProps = (state: GlobalState, ownProps: Props) => {
         lhsOpen: getIsLhsOpen(state),
         rhsOpen: getIsRhsOpen(state),
         rhsMenuOpen: getIsRhsMenuOpen(state),
+        isMobile: state.views.channel.mobileView,
```
Comment (M-ZubairAhmed) [inline components/channel_layout/center_channel/index.ts:48]: would we use `getIsMobileView` selector? 0/5 on it thought

Comment (hmhealey) [reply components/channel_layout/center_channel/index.ts:48]: +1 to that

Comment (jwilander) [reply components/channel_layout/center_channel/index.ts:48]: Yep, good call. I'll update it.

**Review by hmhealey — CHANGES_REQUESTED** [review]
I'm not sure why, but this causes the header to not appear in the onboarding screen for some reason
Screen Shot 2022-02-22 at 11 17 29 AM

**Comment by jwilander** [comment]
@M-ZubairAhmed think you could take this PR over too? I haven't had time to get back to it yet

**Comment by M-ZubairAhmed** [comment]
🧪 e2e report 🧪 

https://automation-dashboard.vercel.app/cycles/01e6ff9f-2cbb-46b1-ab1f-95f45bfe09da

**Comment by M-ZubairAhmed** [comment]
Reopened https://github.com/mattermost/mattermost-webapp/pull/10811

**Pull request closed without merging**